muse_origin/source_creation.py

```diff
@@ -280,8 +281,19 @@ def add_keyword(keyword, param, description, params):
     # Add the FSF information to the source and use this information to compute
     # the PSF weighted spectra.
     if has_fsf:
-        a, b, beta, _ = source.get_FSF()
-        fwhm_fsf = b * data_cube.wave.coord() + a
+        lbda = data_cube.wave.coord()
+        try:
+            a, b, beta, _ = source.get_FSF()
+            fwhm_fsf = b * lbda + a
+        except TypeError:
+            # MPDAF 3.5+
+            model = source.get_FSF()
+            fwhm_fsf = model.get_fwhm(lbda)
+            beta = model.get_beta(lbda)
+            if np.unique(beta).size > 1:
+                raise NotImplementedError
+            beta = beta[0]
```

Comment:
@RolandBacon - I had to do some changes following the latest changes in MPDAF, I think it's OK but please have a look ;)